keycloak-iam/src/main/java/com/valtimo/keycloak/service/KeycloakUserManagementService.java

```diff
@@ -143,7 +145,22 @@ public List<ManageableUser> findByRole(String authority) {
 
     @Override
     public List<ManageableUser> findByRoles(SearchByUserGroupsCriteria groupsCriteria) {
-        return null;
+        Set<String> allUserGroups = new HashSet<>(groupsCriteria.getRequiredUserGroups());
+        groupsCriteria.getOrUserGroups().forEach(allUserGroups::addAll);
+
+        List<ManageableUser> allUsers = new ArrayList<>();
```

Comment:
This could be a direct assignment from the following code.

Reply:
I refactored it a bit